main.py

```diff
@@ -490,7 +490,53 @@ def index(self):
             p.type = point.type
             points.append(p)
         context = {'points': points}
-        return self.render('viewhighlighted.html', **context)
+        return self.render('viewhighlighteddata.html', **context)
+
+    def is_visible(self):
+        return login.current_user.is_authenticated()
+
+class ViewHighlightedMarkersMap(BaseView):
```

Comment:
Also, @SimonMand told me you talked about the way to implement the map in the admin view. What is the best way in your opinion to do that without writing duplicate code? Because most of the relevant code for this is inside index.html, app.js and marker.js.